src/data_egress/sqs_listener.py

```diff
@@ -189,15 +189,37 @@ def get_to_be_processed_s3_prefixes(message):
 
 
 def query_dynamodb(s3_prefixes, dynamodb):
-    """Query  DynamoDb status table for a given correlation id.
+    """Scan DynamoDB table for generic source prefixes.
+
+    Query  DynamoDb status table for a given correlation id.
 
     Arguments:
         s3_prefixes (string): source bucket prefixes to query dynamo db table
     """
+    generic_source_prefixes = []
     table = dynamodb.Table(DATA_EGRESS_DYNAMO_DB_TABLE)
+    scan_response = table.scan(ProjectionExpression=HASH_KEY)
+    for item in scan_response[ITEMS]:
+        if item[HASH_KEY].endswith("*"):
+            logger.info(f"Generic source_prefix found: {item[HASH_KEY]}")
+            generic_source_prefixes.append(item[HASH_KEY].replace("*", ""))
     s3prefix_and_dynamodb_records = []
     for s3_prefix in s3_prefixes:
-        response = table.query(KeyConditionExpression=Key(HASH_KEY).eq(s3_prefix))
+        starts_with_prefix = (
```

Comment:
Appreciate there is nothing there now to build on but maybe a unit test to cover this new behaviour?